sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -174,6 +175,18 @@ private[sql] class HiveSessionCatalog(
   // xpath_double, xpath_float, xpath_int, xpath_long, xpath_number,
   // xpath_short, and xpath_string.
   override def lookupFunction(name: FunctionIdentifier, children: Seq[Expression]): Expression = {
+    try {
+      subLookupFunction(name, children)
+    } catch {
```

Comment:
For the following opinion, I think that is the exact same way of the Spark 1.6 and previous. I think that is not a problem.
> this will fail again as soon as we pass in an argument with a slightly different value